server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMetadataLockManager.java

```diff
@@ -59,15 +60,11 @@ public void acquire(LockInfo lockInfo) throws IOException {
     public void release(LockInfo lockInfo) throws IOException {
         assert lockInfo instanceof FileLockInfo : "lockInfo should be instance of FileLockInfo";
         String[] lockFiles = lockDirectory.listAll();
-
-        // ideally there should be only one lock per acquirer, but just to handle any stale locks,
-        // we try to release all the locks for the acquirer.
-        List<String> locksToRelease = ((FileLockInfo) lockInfo).getLocksForAcquirer(lockFiles);
-        if (locksToRelease.size() > 1) {
-            logger.warn(locksToRelease.size() + " locks found for acquirer " + ((FileLockInfo) lockInfo).getAcquirerId());
-        }
-        for (String lock : locksToRelease) {
-            lockDirectory.deleteFile(lock);
+        try {
+            String lockToRelease = ((FileLockInfo) lockInfo).getLockForAcquirer(lockFiles);
+            lockDirectory.deleteFile(lockToRelease);
+        } catch (NoSuchFileException e) {
+            // Ignoring if the file to be deleted is not present.
```

Comment:
can you add a `info` log as we are ignoring the exception. `No lock file found for acquirerId: {}, lockInfo.getAcquirerId()`. Please update the javaDoc for this method as well to reflect that if file doesn't exist then release will be no-op

Reply:
Added in the revision